x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/OpenAiService.java

```diff
@@ -173,13 +194,19 @@ public OpenAiModel parsePersistedConfig(String inferenceEntityId, TaskType taskT
         Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(config, ModelConfigurations.SERVICE_SETTINGS);
         Map<String, Object> taskSettingsMap = removeFromMapOrDefaultEmpty(config, ModelConfigurations.TASK_SETTINGS);
 
+        ChunkingSettings chunkingSettings = null;
+        if (ChunkingSettingsFeatureFlag.isEnabled() && TaskType.TEXT_EMBEDDING.equals(taskType)) {
```

Comment:
`TaskType.SPARSE_EMBEDDING` can also be chunked.

Reply:
My understanding is that we only offer text embedding for the OpenAIService so adding it to this check might cause some confusion as I don't believe we allow sparse embedding. For other services, I have made sure to check for sparse embedding when it is offered. Let me know if my assumption here is incorrect.

Reply:
True good point